File: csharp-dotnet2/src/main/CsharpDotNet2/IO/Swagger/Model/V3SiriLineRefDirectionRefsDictionary.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace IO.Swagger.Model {

  /// <summary>
  /// 
  /// </summary>
  [DataContract]
  public class V3SiriLineRefDirectionRefsDictionary {
    /// <summary>
    /// Gets or Sets DirectionRefs
    /// </summary>
    [DataMember(Name="direction_refs", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "direction_refs")]
    public Dictionary<string, List<V3SiriReferenceDataDetail>> DirectionRefs { get; set; }

    /// <summary>
    /// Gets or Sets UnmatchedDirectionRefs
    /// </summary>
    [DataMember(Name="unmatched_direction_refs", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "unmatched_direction_refs")]
    public Dictionary<string, string> UnmatchedDirectionRefs { get; set; }


    /// <summary>
    /// Get the string presentation of the object
    /// </summary>
    /// <returns>String presentation of the object</returns>
    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class V3SiriLineRefDirectionRefsDictionary {\n");
      sb.Append("  DirectionRefs: ").Append(DirectionRefs).Append("\n");
      sb.Append("  UnmatchedDirectionRefs: ").Append(UnmatchedDirectionRefs).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }

    /// <summary>
    /// Get the JSON string presentation of the object
    /// </summary>
    /// <returns>JSON string presentation of the object</returns>
    public string ToJson() {
      return JsonConvert.SerializeObject(this, Formatting.Indented);
    }

}
}

File: dart/lib/model/v3_siri_line_ref_direction_refs_dictionary.dart
part of swagger.api;

class V3SiriLineRefDirectionRefsDictionary {
  
  Map<String, List<V3SiriReferenceDataDetail>> directionRefs = {};

  Map<String, String> unmatchedDirectionRefs = {};

  V3SiriLineRefDirectionRefsDictionary();

  @override
  String toString() {
    return 'V3SiriLineRefDirectionRefsDictionary[directionRefs=$directionRefs, unmatchedDirectionRefs=$unmatchedDirectionRefs, ]';
  }

  V3SiriLineRefDirectionRefsDictionary.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    directionRefs = List.mapFromJson(json['direction_refs']);
    unmatchedDirectionRefs = json['unmatched_direction_refs'];
  }

  Map<String, dynamic> toJson() {
    return {
      'direction_refs': directionRefs,
      'unmatched_direction_refs': unmatchedDirectionRefs
     };
  }

  static List<V3SiriLineRefDirectionRefsDictionary> listFromJson(List<dynamic> json) {
    return json == null ? new List<V3SiriLineRefDirectionRefsDictionary>() : json.map((value) => new V3SiriLineRefDirectionRefsDictionary.fromJson(value)).toList();
  }

  static Map<String, V3SiriLineRefDirectionRefsDictionary> mapFromJson(Map<String, Map<String, dynamic>> json) {
    var map = new Map<String, V3SiriLineRefDirectionRefsDictionary>();
    if (json != null && json.length > 0) {
      json.forEach((String key, Map<String, dynamic> value) => map[key] = new V3SiriLineRefDirectionRefsDictionary.fromJson(value));
    }
    return map;
  }
}
